extensions/sql-database-projects/src/test/projectController.test.ts

```diff
@@ -464,103 +449,70 @@ describe('ProjectsController', function (): void {
-			sinon.stub(vscode.window, 'showOpenDialog').resolves(undefined);
-			sinon.stub(vscode.window, 'showQuickPick').resolves({ label: constants.file });
-			const spy = sinon.spy(vscode.window, 'showErrorMessage');
+		it('Callbacks are hooked up and called from create project from database dialog', async function (): Promise<void> {
+			const createProjectFromDbHoller = 'hello from callback for importNewDatabaseProject()';
 
-			const projController = new ProjectsController(new SqlDatabaseProjectTreeViewProvider());
-			await projController.importNewDatabaseProject({ connectionProfile: mockConnectionProfile });
-			should(spy.calledOnce).be.true('showErrorMessage should have been called');
-			should(spy.calledWith(constants.projectLocationRequired)).be.true(`showErrorMessage not called with expected message '${constants.projectLocationRequired}' Actual '${spy.getCall(0).args[0]}'`);
-		});
+			let holler = 'nothing';
 
-		it('Should show error when no location provided with ExtractTarget = SchemaObjectType', async function (): Promise<void> {
-			sinon.stub(vscode.window, 'showInputBox').resolves('MyProjectName');
-			sinon.stub(vscode.window, 'showQuickPick').resolves({ label: constants.schemaObjectType });
-			sinon.stub(vscode.window, 'showOpenDialog').resolves(undefined);
-			const spy = sinon.spy(vscode.window, 'showErrorMessage');
+			const createProjectFromDatabaseDialog = TypeMoq.Mock.ofType(CreateProjectFromDatabaseDialog, undefined, undefined, undefined);
+			createProjectFromDatabaseDialog.callBase = true;
+			createProjectFromDatabaseDialog.setup(x => x.importClick()).returns(async () => {
+				await projController.object.createNewProjectCallBack( { serverId: 'My Id', database: 'My Database', projName: 'testProject', filePath: 'testLocation', version: '1.0.0.0', extractTarget: mssql.ExtractTarget['schemaObjectType'] });
+				return Promise.resolve(undefined);
+			});
 
-			const projController = new ProjectsController(new SqlDatabaseProjectTreeViewProvider());
-			await projController.importNewDatabaseProject({ connectionProfile: mockConnectionProfile });
-			should(spy.calledOnce).be.true('showErrorMessage should have been called');
-			should(spy.calledWith(constants.projectLocationRequired)).be.true(`showErrorMessage not called with expected message '${constants.projectLocationRequired}' Actual '${spy.getCall(0).args[0]}'`);
-		});
+			const projController = TypeMoq.Mock.ofType(ProjectsController);
+			projController.callBase = true;
+			projController.setup(x => x.getCreateProjectFromDatabaseDialog(TypeMoq.It.isAny())).returns(() => createProjectFromDatabaseDialog.object);
+			projController.setup(x => x.createNewProjectCallBack(TypeMoq.It.isAny())).returns(() => {
+				holler = createProjectFromDbHoller;
+				return Promise.resolve(undefined);
+			});
 
-		it('Should set model filePath correctly for ExtractType = File and not-File.', async function (): Promise<void> {
```

Comment:
I didn't see this test over in createProjectFromDatabaseDialog.test.ts; is it no longer necessary?  I remember hitting a regression about this at some point, so I want to make sure it doesn't get lost accidentally.